### PR TITLE
refactor suggest

### PR DESCRIPTION
**Pull request by lionelB**
refactor suggester to a controlled component
add tests

### CODE REVIEW

File: packages/code-du-travail-frontend/src/search/Search.js [hunk 1]
```diff
@@ -208,104 +212,98 @@ class Search extends React.Component {
...
-    if (this.props.onResults && results && results.hits.total > 0) {
-      this.props.onResults(results.hits.hits);
-    }
+
+  onSelect = suggestion => {
+    const { query } = this.state;
+    const route = getRouteBySource(suggestion._source.source);
+    const anchor = suggestion._source.anchor
+      ? suggestion._source.anchor.slice(1)
+      : "";
```
Comment (revolunet) [inline packages/code-du-travail-frontend/src/search/Search.js]: shouldnt it be undefined here ?

File: packages/code-du-travail-frontend/src/search/Search.js [hunk 2]
```diff
@@ -208,104 +212,98 @@ class Search extends React.Component {
...
+
+  onClear = () => {
+    this.setState({ suggestions: [] });
+  };
+
+  onSearch = ({ value }) => {
+    const { excludeSources } = this.state;
+    fetchResultsSuggest(value, excludeSources).then(results => {
+      this.setState({ suggestions: results.hits.hits });
+      this.props.onResults(results);
```
Comment (revolunet) [inline packages/code-du-travail-frontend/src/search/Search.js]: maybe nest the `onResults` call in the `setState` callback ?

File: packages/code-du-travail-frontend/src/search/__tests__/DocumentSuggester.test.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      onSearch
+    });
+
+    const input = getByPlaceholderText(/search foo/i);
+    const mockEvent = { target: { value: "test" } };
+    fireEvent.change(input, mockEvent);
+    expect(onSearch).toBeCalledWith(expect.objectContaining({ value: "test" }));
+  });
+
+  it("should call onSelect", () => {
```
Comment (revolunet) [inline packages/code-du-travail-frontend/src/search/__tests__/DocumentSuggester.test.js]: "when user clicks some option"

**Review by revolunet — APPROVED** [review]

**Comment by revolunet** [comment]
thanks for nice refacto +  testing 👍

File: packages/code-du-travail-frontend/src/search/Search.js (hunk 1 above)
Comment (lionelB) [reply packages/code-du-travail-frontend/src/search/Search.js]: hum probably but I need to test :)

File: packages/code-du-travail-frontend/src/search/Search.js (hunk 2 above)
Comment (lionelB) [reply packages/code-du-travail-frontend/src/search/Search.js]: I see that pattern already but I don't know why ? this defer the callback ?

Comment (revolunet) [reply packages/code-du-travail-frontend/src/search/Search.js]: yes, this ensure the order is respected by triggering the callback only when state is effectively updated.

**Pull request merged**